### ISSUE TITLE
Tests need to fail gracefully without internet

### ISSUE THREAD

**Issue by cboettig** [ropensci/codemetar#230]
A good number of functions require an internet connection, running the test suite without a connection still goes poorly.  CRAN's solaris machines seem particularly good at failing to make successful internet calls and CRAN doesn't like it when this causes the tests to fail...

**Comment by maelle** [comment]
So should the whole test suite be skipped if there's no internet connection? Not very graceful though, we should make the whole package better at working w/o internet. 🤔

**Comment by maelle** [comment]
Cf also #175

**Comment by cboettig** [comment]
I modified `codemeta_validate()` to return with a warning if no internet connection is available, and for `sys_reqs` method to return NULL if no connection is available.  Then I decided that patching everything wasn't going to happen before CRAN kicked us off and switched over to skipping tests instead.  

I used `pingr::is_online()` to define a `skip_if_offline()` method (which is apparently already available as a skip in the dev version of `testthat`, though not sure if it uses `pingr` or something else).  I added that skip to any `test_that()` block that failed when I switched internet off (i.e. most but not all).  

In principle we should overhaul the design to work better in offline mode (i.e. warnings and fall-back methods instead of errors), though in practice that loses a lot of the richness of the current package.  A codemeta-lite that worked with minimal dependencies to parse local DESCRIPTION, README, etc would be a more practical offline solution and could eventually be swapped in here to provide the core functionality.  but that's a big lift so not sure when it would happen.
